payments-core/src/main/java/com/stripe/android/googlepaylauncher/GooglePayLauncherActivity.kt

```diff
@@ -0,0 +1,216 @@
+package com.stripe.android.googlepaylauncher
+
+import android.content.Intent
+import android.content.pm.ActivityInfo
+import android.os.Build
+import android.os.Bundle
+import androidx.activity.viewModels
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.lifecycleScope
+import com.google.android.gms.wallet.AutoResolveHelper
+import com.google.android.gms.wallet.PaymentData
+import com.google.android.gms.wallet.PaymentDataRequest
+import com.google.android.gms.wallet.PaymentsClient
+import com.stripe.android.model.PaymentMethodCreateParams
+import com.stripe.android.model.StripeIntent
+import com.stripe.android.view.AuthActivityStarterHost
+import kotlinx.coroutines.launch
+import org.json.JSONObject
+
+/**
+ * [GooglePayLauncherActivity] is used to return the result of a Google Pay operation.
+ * The activity's result will be a [GooglePayLauncher.Result].
+ *
+ * [GooglePayLauncherActivity] will:
+ * 1. Fetch the [StripeIntent]
+ * 2. Verify that Google Pay is available
+ * 3. Create a [PaymentDataRequest](https://developers.google.com/pay/api/web/reference/request-objects#PaymentDataRequest)
+ * 4. Confirm the [StripeIntent] using the Google Pay token
+ *
+ * Use [GooglePayLauncherContract] to start [GooglePayLauncherActivity].
+ *
+ * See [Troubleshooting](https://developers.google.com/pay/api/android/support/troubleshooting)
+ * for a guide to troubleshooting Google Pay issues.
+ */
+internal class GooglePayLauncherActivity : AppCompatActivity() {
```

Comment:
I am curious where you ran into problems reusing code from `StripeGooglePayActivity` and `StripeGooglePayViewModel`

Reply:
There were enough differences between how `StripeGooglePayActivity` works and how `GooglePayLauncherActivity` should work that I decided to reconcile the differences later. For example, `GooglePayLauncherActivity` confirms a Payment Intent - `StripeGooglePayActivity` just returns a `PaymentMethod`.